samcli/commands/_utils/click_mutex.py

```diff
@@ -0,0 +1,89 @@
+"""
+Module to check mutually exclusive cli parameters
+"""
+from typing import Any, List, Dict, Tuple
+
+import click
+
+
+class ClickMutex(click.Option):
+    """
+    Preprocessing checks for mutually exclusive or required parameters as supported by click api.
+
+    required_param_lists: List[List[str]]
+        List of lists with each supported combination of params
+        Ex:
+        With option = "a" and required_param_lists = [["b", "c"], ["c", "d"]]
+        It is valid to specify --a --b --c or --a --c --d
+        but not --a --b --d
+
+    required_params_hint: str
+        String to be appended after default missing required params prompt
+
+    incompatible_params: List[str]
+        List of incompatible parameters
+
+    incompatible_params_hint: str
+        String to be appended after default incompatible params prompt
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.required_param_lists: List = kwargs.pop("required_param_lists", [])
```

Comment:
nit: `self.required_param_lists` and `self.incompatible_params` seem to follow different naming patterns.